s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/ProductRepository.java

```diff
@@ -13,19 +13,30 @@
 public interface ProductRepository extends CrudRepository<Product, Long> {
     <T> List<T> findAllBy(Class<T> projection);
 
-    @Query("SELECT CASE WHEN COUNT(p)> 0 THEN 'true' ELSE 'false' END FROM Product p " +
-            "JOIN p.favourites f " +
-            "WHERE p.id = :productId AND f.email = :email")
-    boolean isFavouriteByEmailAndProductId(String email, Long productId);
+    @Query("SELECT p " +
+            "FROM Product p ")
```

Comment:
In fact using newline here doesn't make sense.

However, `@Query("SELECT p FROM Product p")` makes sense, as it's not a native query but JPQL, which in its abstraction operates on objects.

Specifiying the query is unfortunately necessary, as I fetch schemas in it. I could annotate `findAll` method with `@EntityGraph`, but then all calls to it would unnecessarily fetch schemas.

This behaviour is really something I don't like about using repositories in general, but having stuck with them I don't see a more sensible way to have different fetch graphs and, at the same time, don't pollute general `findAll` method with that. The graph you use should actually depend on the projection you use, but I haven't found a way to specify it on a call level.

Reply:
ok